fuzz-simplifier failure

Unrelated PR https://github.com/halide/Halide/pull/3995 happened happened to come up with a fuzz-simplify failure in https://buildbot.halide-lang.org/master/#/builders/26/builds/111 👍 

```
FAILED TEST: fuzz_simplify
Simplify fuzz test seed: 1562913093
a = uint8(a)
b = uint8(a)
c = (uint8)85
d = (uint8)165
e = (uint8)100
absd(ramp(uint8(c), uint8(a), 2), select((min((uint8)230, min(uint8(e), (uint8(c) + (uint8)82))) <= absd((uint8)84, (uint8)145)), ((uint8x2(x2(uint16(d))) + (x2(uint8(c))*ramp((uint8)223, uint8(b), 2))) - ((x2(uint8(e))/ramp(uint8(e), uint8(a), 2)) % (x2((uint8)73) % ramp((uint8)53, uint8(d), 2)))), ((x2(uint8(d)) - x2(uint8(d)))/(ramp(uint8(c), uint8(b), 2) - x2(uint8(c))))))
absd(ramp(uint8(c), uint8(a), 2), select(((uint8)61 < min(min((uint8(c) + (uint8)82), uint8(e)), (uint8)230)), x2((uint8)0), (ramp(((uint8(c)*(uint8)223) + uint8(d)), (uint8(b)*uint8(c)), 2) - ((x2(uint8(e))/ramp(uint8(e), uint8(a), 2)) % (x2((uint8)73) % ramp((uint8)53, uint8(d), 2))))))
In vector lane 0:
absd(uint8(c), (((uint8(c)*(uint8)223) + uint8(d)) - (uint8)1)) -> (uint8)90
absd(uint8(c), select(((uint8)61 < min(min((uint8(c) + (uint8)82), uint8(e)), (uint8)230)), (uint8)0, (((uint8(c)*(uint8)223) + uint8(d)) - (uint8)1))) -> (uint8)85
../halide/Makefile:1828: recipe for target 'quiet_correctness_fuzz_simplify' failed
```

--- a/src/Simplify_Div.cpp
+++ b/src/Simplify_Div.cpp
@@ -80,6 +80,12 @@ Expr Simplify::visit(const Div *op, ExprInfo *bounds) {
         bounds->trim_bounds_using_alignment();
     }
 
+    bool denominator_non_zero =
+        ((b_bounds.min_defined && b_bounds.min > 0) ||
+         (b_bounds.max_defined && b_bounds.max < 0) ||
+         (b_bounds.alignment.remainder != 0));
+
+
     if (may_simplify(op->type)) {
 
         int lanes = op->type.lanes();
@@ -93,8 +99,9 @@ Expr Simplify::visit(const Div *op, ExprInfo *bounds) {
             rewrite(x / 1, x) ||
             (!op->type.is_float() &&
              rewrite(x / 0, IRMatcher::Indeterminate())) ||
-            rewrite(0 / x, 0) ||
-            rewrite(x / x, 1) ||
+            (denominator_non_zero &&
+             (rewrite(0 / x, 0) ||
+              rewrite(x / x, 1))) ||
             rewrite(c0 / c1, fold(c0 / c1))) {
             return rewrite.result;
         }

--- a/test/correctness/simplify.cpp
+++ b/test/correctness/simplify.cpp
@@ -233,7 +233,9 @@ void check_algebra() {
 
     check(0/max(x, 1), 0);
     check(x/1, x);
-    check(x/x, 1);
+    check(max(x, 1)/(max(x, 1)), 1);
+    check(min(x, -1)/(min(x, -1)), 1);
+    check((x*2+1)/(x*2+1), 1);
     check((-1)/(x*2 + 1), select(x < 0, 1, -1));
     check(Expr(7)/3, 2);
     check(Expr(6.0f)/2.0f, 3.0f);

